src/indexing.jl

```diff
@@ -17,12 +24,9 @@ struct IndexEntry
     tags::NTuple{<:Any, Pair{Symbol, <:AbstractString}}
 end
 
-IndexEntry(file) = IndexEntry(Path(file))
-function IndexEntry(filepath::AbstractPath)
-    # skip any non-tag directories at the start. Note this will be tricked if those have "="
-    # in them but probably not worth handling, unless an issue comes up
-    first_tag_ind = something(findfirst(contains("="), filepath.segments), 1)
-    segments = filepath.segments[first_tag_ind:end-1]
+IndexEntry(file, base_dir) = IndexEntry(Path(file), Path(base_dir))
```

Comment:
This is not tested but I notice that in general we aren't testing `IndexEntry` directly.

Might be orthogonal to changes, but why do we have this anyway? It looks like `index_checkpoint_files` will always pass in `AbstractPath`.

Reply:
correct.
Lets delete it

```suggestion
```